### PR TITLE
feat(rfs/hook): add new resource to manage private hook

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Supports a new resource to manage private hook.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource to manage private hook.
```

## PR Checklist

* [x] Tests added/passed.

```
NONE
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

### CODE REVIEW

File: docs/resources/rfs_private_hook.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region where the private hook is located.  
+  If omitted, the provider-level region will be used. Change this parameter will create a new resource.
+
+* `name` - (Required, String, ForceNew) Specifies the name of the private hook.  
+  The valid length is limited from can contain `1` to `128`, only Chinese or English letters, digits, hyphens (-),
```
Comment (deer-hang) [inline docs/resources/rfs_private_hook.md]:
```suggestion
  The valid length is limited from `1` to `128`, only Chinese or English letters, digits, hyphens (-),
```

File: docs/resources/rfs_private_hook.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+```hcl
+resource "huaweicloud_rfs_private_hook" "test" {
+  ...
+
+  lifecycle {
+    ignore_changes = [
+      policy_uri,
+      keep_old_version,
+    ]
```
Comment (deer-hang) [inline docs/resources/rfs_private_hook.md:163]:
```suggestion
    policy_body]
```

File: huaweicloud/services/rfs/resource_huaweicloud_rfs_private_hook.go [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+		MoreHeaders: map[string]string{
+			"Content-Type":      "application/json",
+			"Client-Request-Id": requestId,
+		},
+		JSONBody: utils.RemoveNil(buildPrivateHookCreateBodyParams(d)),
+	}
+
+	requestResp, err := client.Request("POST", createPath, &opt)
+	if err != nil {
+		return diag.Errorf("error create private hook: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/rfs/resource_huaweicloud_rfs_private_hook.go]:
```suggestion
		return diag.Errorf("error creating private hook: %s", err)
```

File: huaweicloud/services/rfs/resource_huaweicloud_rfs_private_hook.go [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+			}
+		}
+		requestBody["default_version"] = newVersion.(string)
+	} else if d.HasChanges("version_description", "policy_uri", "policy_body") {
+		return diag.Errorf("Unable to update version description, policy URI or policy body without updating version")
+	}
+
+	err = updatePrivateHookMetadata(client, hookName, requestBody)
+	if err != nil {
+		diag.Errorf("error updating private hook (%s): %s", hookName, err)
```
Comment (deer-hang) [inline huaweicloud/services/rfs/resource_huaweicloud_rfs_private_hook.go]:
```suggestion
		return diag.Errorf("error updating private hook (%s): %s", hookName, err)
```

File: docs/resources/rfs_private_hook.md (hunk 2 above)
Comment (Lance52259) [reply docs/resources/rfs_private_hook.md:163]: only one argument can be configured at the same time

File: docs/resources/rfs_private_hook.md (hunk 1 above)
Comment (Lance52259) [reply docs/resources/rfs_private_hook.md]: add incorrect descriptions are fixed.

File: huaweicloud/services/rfs/resource_huaweicloud_rfs_private_hook.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/rfs/resource_huaweicloud_rfs_private_hook.go]: done.

File: huaweicloud/services/rfs/resource_huaweicloud_rfs_private_hook.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/rfs/resource_huaweicloud_rfs_private_hook.go]: done.

**Pull request merged**